wagtail/search/backends/database/fallback.py

```diff
@@ -56,6 +56,9 @@ def _process_lookup(self, field, lookup, value):
             **{field.get_attname(self.queryset.model) + "__" + lookup: value}
         )
 
+    def _process_match_none(self):
+        return models.Q(pk__in=[])
```

Comment:
I had the same idea, but wasn't sure whether to put this in `_connect_filters`, `_process_filters`, or a new method altogether. Glad to see the same approach!